control-plane/connect-inject/controllers/endpoints/endpoints_controller_test.go

```diff
@@ -6414,3 +6414,54 @@ func createGatewayPod(name, ip string, annotations map[string]string) *corev1.Po
 	}
 	return pod
 }
+
+func TestReconcileAssignServiceVirtualIP(t *testing.T) {
+	t.Parallel()
+	ctx := context.Background()
+	cases := []struct {
```

Comment:
Is there no way to check if `apiClient.Internal().AssignServiceVirtualIP(ctx, svc.Service, []string{IP}, &api.WriteOptions{})` executed correctly? as in, can we assert against the desired outcome?

Reply:
I don't think so. We are operating blind here as there is no API endpoint and/or anything usable in the API/SDK. The VIP also ends up buried in the discovery chain...